### ISSUE TITLE
classes prop instead of className

### ISSUE THREAD

**Issue by roti** [facebook/react#24515]
It is often the case that multiple CSS classes are used for an element, and sometimes they are dynamically generated. In such situations the client code needs to concatenate them into a string
```
const classes = getCSSClasses();
return (
    <div className={classes.class1 + " " + classes.class2}> foo </div>
);
```

I always felt the need to be able to pass the classes as an array:
```
const classes = getCSSClasses();
return (
    <div classes={[classes.class1, classes.class2]}> foo </div>
);
```

In addition to the convenience of such a property, maybe it could help with performance, in the sense that equality could be checked before string concatenation, and only do the concatenation if the element needs to be rendered.

The drawback here is of course the mutable nature of an array. I imagine performance would be better if the array would be immutable. So maybe this new property could work only with some immutable array implementation. Or frozen arrays.

In short, I want such a property for convenience, but maybe it also has potential for some performance improvements.

**Comment by kkyusufk** [comment]
Not sure if this might help, but there is a very good library for what you are describing
https://www.npmjs.com/package/classnames

**Comment by arthuralcorrea** [comment]
> Not sure if this might help, but there is a very good library for what you are describing
> https://www.npmjs.com/package/classnames

Perhaps this library it's widely used, I also miss a better built in approach to handle CSS classes.

Maybe a solution inspired in this library will be a good thing.

**Comment by roti** [comment]
To be honest, convenience is probably not enough as an argument to add such an API. It's true, it can be faily easily solved in more than one way.
I am bothered by this string concatenation, which necessary because of the DOM API, and the browser has then to split it again. Maybe the question is how much performance we loose because of it. Ideally the prop should be a collection of class names, and the DOM Api should accept a collection as well, instead of a concatenated string.

**Comment by eps1lon** [comment]
Duplicate of https://github.com/facebook/react/issues/10271. See https://github.com/facebook/react/issues/10271#issuecomment-334276806 for why this was closed.
